components/scream/src/share/grid/remap/coarsening_remapper.cpp

```diff
@@ -0,0 +1,1002 @@
+#include "coarsening_remapper.hpp"
+
+#include "share/grid/point_grid.hpp"
+#include "share/io/scorpio_input.hpp"
+
+#include <ekat/kokkos/ekat_kokkos_utils.hpp>
+#include <ekat/ekat_pack_utils.hpp>
+
+#include <numeric>
+
+namespace scream
+{
+
+CoarseningRemapper::
+CoarseningRemapper (const grid_ptr_type& src_grid,
+                    const std::string& map_file)
+ : AbstractRemapper()
+ , m_comm (src_grid->get_comm())
+{
+  using namespace ShortFieldTagsNames;
+
+  // Sanity checks
+  EKAT_REQUIRE_MSG (src_grid->type()==GridType::Point,
+      "Error! CoarseningRemapper only works on PointGrid grids.\n"
+      "  - src grid name: " + src_grid->name() + "\n"
+      "  - src_grid_type: " + e2str(src_grid->type()) + "\n");
+  EKAT_REQUIRE_MSG (src_grid->is_unique(),
+      "Error! CoarseningRemapper requires a unique source grid.\n");
+
+  // This is a coarsening remapper. We only go in one direction
+  m_bwd_allowed = false;
+
+  // Create io_grid, containing the indices of the triplets
+  // in the map file that this rank has to read
+  auto gids_h = get_my_triplets_gids (map_file,src_grid);
+  view_1d<gid_t> gids_d("",gids_h.size());
+  Kokkos::deep_copy(gids_d,gids_h);
+
+  auto io_grid = std::make_shared<PointGrid>("",gids_h.size(),0,m_comm);
+  io_grid->set_dofs(gids_d);
+  const int nlweights = io_grid->get_num_local_dofs();
+
+  // Create CRS matrix views
+
+  // Read in triplets.
+  view_1d<gid_t>::HostMirror  row_gids_h("",nlweights);
+  view_1d<gid_t>::HostMirror  col_gids_h("",nlweights);
+  view_1d<double>::HostMirror S_h ("",nlweights);
+
+  // scream's gids are of type int, while scorpio wants long int as offsets.
+  std::vector<scorpio::offset_t> dofs_offsets(nlweights);
+  for (int i=0; i<nlweights; ++i) {
+    dofs_offsets[i] = gids_h[i];
+  }
+  const std::string idx_decomp_tag = "int-nnz" + std::to_string(nlweights);
```

Comment:
I thought including the nnz value was enough, but in the remote case that the nnz read at the beginning (when each rank has yet to figure out which triplets it needs) and the nnz read here have are the same, it would not be enough. I'll change it.